common/persistence/src/main/java/app/coronawarn/server/common/persistence/service/TraceTimeIntervalWarningService.java

```diff
@@ -71,4 +109,20 @@ public int saveCheckinData(List<CheckIn> checkins) {
             .findAll(Sort.by(Direction.ASC, "submissionTimestamp")).iterator())
         .collect(Collectors.toList());
   }
-}
+
+  private byte[] hashLocationId(ByteString locationId) {
+    try {
+      return MessageDigest.getInstance("SHA-256").digest(locationId.toByteArray());
+    } catch (NoSuchAlgorithmException e) {
+      logger.warn("Could not apply SHA-256 hash on " + locationId
+          + " while storing TraceTimeIntervalWarnings");
+    }
+    return NO_HASH;
```

Comment:
Proposal: Would prefer to calculate this for the whole batch at once. If one fails: raise an exception, raise the log level to error and reject the whole submission altogether.